src/components/CourseSetEdit.vue

```diff
@@ -12,70 +12,83 @@
+          v-for="courseSet in Object.keys(getCourseSets)"
+          :key="courseSet"
         >
-        <b-form-invalid-feedback>
-          Must be a unique name
-        </b-form-invalid-feedback>
-        <!-- I dont actually show any form valid feedback, but having this here keeps
-           The page nicely spaced out and not bouncing-->
-        <b-form-valid-feedback id="valid-feedback">
-          Must be a unique name
-        </b-form-valid-feedback>
-      </b-input-group>
+          <font-awesome-icon
+            v-if="Object.keys(getCourseSets).length > 1"
+            :icon="['fas', 'trash']"
+            class="open_close_icon, trash-btn"
+            @click="removeCourseSet(courseSet)"
+          ></font-awesome-icon>
+          {{ courseSet }}
+        </li>
+        <li class="list-group-item">
+          <b-input-group>
+            <b-form-input
+              v-model="newCourseSetName"
+              :state="newCourseSetExists"
+              placeholder="Course Set Name"
+              aria-lable="Course Set Name"
+              trim
+              @keyup.enter="createNewCourseSet"
+            ></b-form-input>
+            <b-input-group-append>
+              <b-button
+                @click="createNewCourseSet"
+                style="
+                  border-top-right-radius: 0.25rem;
+                  border-bottom-right-radius: 0.25rem;
+                "
+                :disabled="!newCourseSetExists"
+                :class="{
+                  'btn-success': newCourseSetExists,
+                }"
+                :title="newCourseSetExists ? '' : 'Disabled'"
+                >Add Course Set</b-button
+              ></b-input-group-append
+            >
+            <b-form-invalid-feedback>
+              Must be a unique name
+            </b-form-invalid-feedback>
+            <!-- I dont actually show any form valid feedback, but having this here keeps
+               The page nicely spaced out and not bouncing-->
+            <b-form-valid-feedback id="valid-feedback">
+              Must be a unique name
+            </b-form-valid-feedback>
+          </b-input-group>
 
-      <h3 style="margin: 0px;">Current course sets:</h3>
-      <p style="font-size: 80%;" v-if="Object.keys(getCourseSets).length <= 1">
-        There must always be 1 course set, add another to remove the current
-        course set
-      </p>
-      <div v-for="courseSet in Object.keys(getCourseSets)" :key="courseSet">
-        <font-awesome-icon
-          v-if="Object.keys(getCourseSets).length > 1"
-          :icon="['fas', 'trash']"
-          class="open_close_icon, trash-btn"
-          @click="removeCourseSet(courseSet)"
-        ></font-awesome-icon>
-        {{ courseSet }}
-      </div>
+          <p
+            style="font-size: 80%;"
+            v-if="Object.keys(getCourseSets).length <= 1"
+          >
+            There must always be 1 course set, add another to remove the current
```

Comment:
Is this still needed?  The deletion icon only appears once there's at least two course sets.